### PR TITLE
core(metrics): add lantern speed index

### LINKED ISSUE #4639: Investigate Lantern predicting Speed Index
For the new Lighthouse performance score upgrade, we want to use Speed Index as a mid-load predictor. To make our story consistent, we need to make sure Lantern is able to predict this as well.

### PR DESCRIPTION
**Pull request by patrickhulce**
closes #4639 
blocked from merging on #4966 

predicts speed index by using the real speed index as the optimistic estimate and a layout-based speed index approximation from the pessimistic TTI graph as the pessimistic estimate

accuracy is decent. probably wouldn't recommend making it a headline metric as-is, but correlates slightly better than FCP (less than TTI/TTFCPUI)

spearman's rho ~.78, MAPE ~47%

### CODE REVIEW

**Comment by patrickhulce** [comment]
OK folks FYI I've updated this and it's ready for review!

**Comment by patrickhulce** [comment]
friendly reminder here that I/O-Expedite policy has this currently auto-merging at 2pm today folks ;)

File: lighthouse-core/gather/computed/metrics/lantern-speed-index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    metricResult.timing = Math.max(metricResult.timing, fcpResult.timing);
+    return metricResult;
+  }
+
+  /**
+   * @param {Map<Node, LH.Gatherer.Simulation.NodeTiming>} nodeTiming
+   * @param {number} fcpTimeInMs
+   * @return {number}
+   */
+  static computeLayoutBasedSpeedIndex(nodeTiming, fcpTimeInMs) {
```
Comment (paulirish) [inline lighthouse-core/gather/computed/metrics/lantern-speed-index.js]: can you add some documentation that explains the approach here?

Comment (patrickhulce) [reply lighthouse-core/gather/computed/metrics/lantern-speed-index.js]: 👍 done

**Pull request merged**